ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/io/FileOperations.java

```diff
@@ -0,0 +1,115 @@
+/*
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package org.apache.hadoop.hive.ql.parse.repl.dump.io;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.LocatedFileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.RemoteIterator;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.ReplChangeManager;
+import org.apache.hadoop.hive.ql.exec.ReplCopyTask;
+import org.apache.hadoop.hive.ql.parse.EximUtil;
+import org.apache.hadoop.hive.ql.parse.ReplicationSpec;
+import org.apache.hadoop.hive.ql.parse.SemanticException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+
+public class FileOperations {
+  private static Logger logger = LoggerFactory.getLogger(FileOperations.class);
+  private final Path dataFileListPath;
+  private final Path exportRootDataDir;
+  private HiveConf hiveConf;
+  private final FileSystem dataFileSystem, exportFileSystem;
+
+  public FileOperations(Path dataFileListPath, Path exportRootDataDir, HiveConf hiveConf)
+      throws IOException {
+    this.dataFileListPath = dataFileListPath;
+    this.exportRootDataDir = exportRootDataDir;
+    this.hiveConf = hiveConf;
+    dataFileSystem = dataFileListPath.getFileSystem(hiveConf);
+    exportFileSystem = exportRootDataDir.getFileSystem(hiveConf);
+  }
+
+  public void export(ReplicationSpec forReplicationSpec) throws IOException, SemanticException {
+    if (forReplicationSpec.isLazy()) {
+      exportFilesAsList();
+    } else {
+      copyFiles();
+    }
+  }
+
+  /**
+   * This writes the actual data in the exportRootDataDir from the source.
+   */
+  private void copyFiles() throws IOException {
+    RemoteIterator<LocatedFileStatus> itr = dataFileSystem.listFiles(dataFileListPath, true);
+    while (itr.hasNext()) {
+      LocatedFileStatus fileStatus = itr.next();
+      if (shouldExport(fileStatus)) {
+        ReplCopyTask.doCopy(exportRootDataDir, exportFileSystem, fileStatus.getPath(), dataFileSystem, hiveConf);
```

Comment:
yeh it would be more work to just partially move the _files creation here and leave the copy there, since the larger goal is to move everything here, it was easier to do the same.